src/Symfony/Component/DependencyInjection/Util/ReflectionHelper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Util;
+
+use Go\ParserReflection\ReflectionClass;
+
+/**
+ * ReflectionHelper provides method for native or static reflection of class.
+ *
+ * @author Martin Hasoň <martin.hason@gmail.com>
+ */
+class ReflectionHelper
+{
+    private static $reflections = array();
+
+    /**
+     * Returns reflection instance for class.
+     *
+     * @param $class Full name of the class
+     *
+     * @return \ReflectionClass
+     */
+    public function getReflection($class)
+    {
+        if (isset(self::$reflections[$class])) {
+            return self::$reflections[$class];
+        }
+
+        if (class_exists($class, false) || interface_exists($class, false) || trait_exists($class, false)) {
+            return $this->getNativeReflection($class);
+        }
+
+        if (class_exists(ReflectionClass::class)) {
+            return $this->getGoParserReflection($class);
+        }
+
+        return $this->getNativeReflection($class);
+    }
+
+    private function getNativeReflection($class)
+    {
+        try {
+            return self::$reflections[$class] = new \ReflectionClass($class);
+        } catch (\ReflectionException $reflectionException) {
+        }
+    }
+
+    private function getGoParserReflection($class)
+    {
+        try {
+            $reflectionClass = new ReflectionClass($class);
+            $reflectionClass->__toString(); // checks that reflection is complete and valid
```

Comment:
Personally, I don't like this trick with `__toString`, it will be better to call `getParentClass()` + `getInterfaces()` + `getTraits()` instead to verify all this stuff with unavailable parent classes.